tests/manage/z_cluster/nodes/test_nodes_restart_ms.py

```diff
@@ -0,0 +1,180 @@
+
+        osd_node_name = random.choice(get_osd_running_nodes())
+        osd_node = get_node_objs([osd_node_name])[0]
+
+        old_osd_pod_ids = get_node_osd_ids(osd_node_name)
+        logger.info(f"osd pod ids: {old_osd_pod_ids}")
+        node_osd_pods = pod.get_osd_pods_having_ids(old_osd_pod_ids)
+        node_osd_pod_names = [p.name for p in node_osd_pods]
+
+        logger.info(f"Going to restart the node {osd_node_name}")
+        nodes.restart_nodes(nodes=[osd_node], wait=False)
+
+        logger.info("Verify the node osd pods go into a Terminating state")
+        res = pod.wait_for_pods_to_be_in_statuses(
+            [constants.STATUS_TERMINATING], node_osd_pod_names
+        )
+        assert res, "Not all the node osd pods are in a Terminating state"
+
+        wait_for_nodes_status(node_names=[osd_node_name])
+        assert wait_for_osd_ids_come_up_on_node(
+            osd_node_name, old_osd_pod_ids, timeout=300
+        )
+        logger.info(
+            f"the osd ids {old_osd_pod_ids} Successfully come up on the node {osd_node_name}"
+        )
+
+        logger.info("Verify the worker nodes security groups on the provider...")
+        assert verify_worker_nodes_security_groups()
+
+    @tier4a
+    @pytest.mark.parametrize(
+        argnames=["node_type"],
+        argvalues=[
+            pytest.param(
+                *[constants.WORKER_MACHINE], marks=pytest.mark.polarion_id("OCS-3982")
+            ),
+            pytest.param(
+                *[constants.MASTER_MACHINE], marks=pytest.mark.polarion_id("OCS-3981")
+            ),
+        ],
+    )
+    def test_nodes_restart(self, nodes, node_type):
+        """
+        Test nodes restart (from the platform layer)
+
+        """
+        if node_type == constants.WORKER_MACHINE:
+            node_names = get_worker_nodes()
+        else:
+            node_names = get_master_nodes()[0:2]
+
+        ocp_nodes = get_node_objs(node_names=node_names)
+        nodes.restart_nodes(nodes=ocp_nodes, wait=False)
+        self.sanity_helpers.health_check()
+        self.create_resources()
+
+    @tier4b
+    @bugzilla("1754287")
+    @pytest.mark.polarion_id("OCS-2015")
+    @pytest.mark.parametrize(
+        argnames=["node_type"],
+        argvalues=[
+            pytest.param(constants.WORKER_MACHINE),
+            pytest.param(constants.MASTER_MACHINE),
+        ],
+    )
+    def test_rolling_nodes_restart(self, nodes, node_type):
+        """
+        Test restart nodes one after the other and check health status in between
+
+        """
+        if node_type == constants.WORKER_MACHINE:
+            node_names = get_worker_nodes()
+        else:
+            node_names = get_master_nodes()
+
+        ocp_nodes = get_node_objs(node_names=node_names)
+        for node in ocp_nodes:
+            nodes.restart_nodes(nodes=[node], wait=False)
+            self.sanity_helpers.health_check(cluster_check=False, tries=60)
+
+        self.create_resources()
```

Comment:
the implementation seems quite similar to the existing test case of non-MS. Please move similar implementation to a common function in order to avoid code duplication

Reply:
Yes, but the created resources should be different when we use the MS. Also, it is slightly different because I roll restart the worker nodes the first time and the master nodes the second time when using MS. 
Anyway, I will change the first lines to be more clear.